src/common/status.h

```diff
@@ -168,7 +168,7 @@ struct StringInStatusOr<T, std::enable_if_t<sizeof(T) < sizeof(std::string)>> :
   StringInStatusOr(StringInStatusOr<U>&& v) : BaseType(new std::string(*std::move(v))) {}  // NOLINT
   template <typename U, typename std::enable_if_t<!StringInStatusOr<U>::inplace, int> = 0>
   StringInStatusOr(StringInStatusOr<U>&& v)  // NOLINT
-      : BaseType((typename StringInStatusOr<U>::BaseType &&)(std::move(v))) {}
+      : BaseType((typename StringInStatusOr<U>::BaseType&&)(std::move(v))) {}
```

Comment:
Should revert this file. It seems the different behaviors between different versions of clang-format. For clang-format >= 14, it will add a space for type and parameter name, but not before clang-format 14.

Reply:
We can add a clang-format rule for this? Seems check by clang-tidy version is not a good way.

Reply:
Ok, I will check the versions and will revert it.

Reply:
I reverted the changes. It was because I use clang-format version 17.*.

Should I investigate this a bit more and try to add a rule?


Also, I'm trying to run linter with the command (actually each time I try to run linter it fails):
```bash
$ ./x.py check tidy --clang-tidy-path /.../llvm@14/bin/clang-tidy --run-clang-tidy-path /.../llvm@14/bin/run-clang-tidy
```

and receive an output:

```
4975 warnings and 1 error generated.
Error while processing kvrocks/src/types/bloom_filter.cc.
Suppressed 4974 warnings (4966 in non-user code, 8 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
9 warnings treated as errors
Traceback (most recent call last):
  File "./x.py", line 422, in <module>
    args.func(**arg_dict)
  File "./x.py", line 195, in clang_tidy
    run(run_command, *options, *regexes, verbose=True, cwd=basedir)
  File "./x.py", line 68, in run
    raise RuntimeError(err)
RuntimeError:
failed to run: ('/opt/homebrew/opt/llvm@14/bin/run-clang-tidy', '-p', 'build', '-clang-tidy-binary', '/opt/homebrew/opt/llvm@14/bin/clang-tidy', '-j8', '-header-filter=kvrocks/src/|utils/kvrocks2redis/|tests/cppunit/', 'kvrocks/src/', 'utils/kvrocks2redis/', 'tests/cppunit/')
exit with code: 1
```

And the problem is, I guess that it's not a normal behavior, and I can't see an actual error, or if it actually exists. Are there some suggestions which I can try to resolve this?

Reply:
Sure, you can assign me to an issue and I will do it ASAP.